packages/core/src/extensions/Blocks/PreviousBlockTypePlugin.ts

```diff
@@ -8,6 +8,13 @@ import { Decoration, DecorationSet } from "prosemirror-view";
 
 const PLUGIN_KEY = new PluginKey(`previous-blocks`);
 
+// Used to change HTML attribute string from camelCase to kebab-case to better follow naming conventions.
+const changeAttributeCase = (str: string) =>
```

Comment:
why do we need to changeAttributeCase? attributes will never be camelcase anymore right?

Reply:
TipTap node attributes still use camelCase as a convention (makes sense since hyphens are reserved symbols in JS), so they only get changed to kebab-case when rendering them to HTML (also because we prepend custom HTML attributes with `data-*`). E.g. `headingType` gets rendered out to `data-heading-type`.

TipTap internal representation:
```
{
  "type": "block",
  "attrs": {
    **"headingType": "1"**
  },
  "content": [
    {
      "type": "content",
      "attrs": {},
      "content": [
        {
          "type": "text",
          "text": "Heading"
        }
      ]
    }
  ]
}
```

Rendered HTML:
```
<div data-heading-type="1" class="..." data-node-type="block-outer">
  <div data-heading-type="1" class="..." data-node-type="block">
    <div class="..." data-node-type="block-content">
      <div>
        This is a heading
      </div>
    </div>
  </div>
</div>
```

In hindsight I should have used a regex check for all HTML attribute rendering & parsing in `Block.tsx` since it's cleaner than mapping TipTap node attributes to HTML custom attributes, but either approach works.